uwp/uwp_main.cpp

```diff
@@ -421,12 +378,89 @@ void App::Run()
 void App::Uninitialize()
 {
 	main_exit(NULL);
+	
+	//if this instance of RetroArch was started from another app/frontend and the frontend passed "launchOnExit" parameter:
+	//1. launch the app specified in "launchOnExit", most likely the same app that started RetroArch
+	//2. RetroArch goes to background and RunAsyncAndCatchErrors doesn't return, because the target app is immediately started.
+	//3. explicitly exit in App::OnEnteredBackground if m_launchOnExitShutdown is set. Otherwise, RetroArch doesn't properly shutdown.
+	if (m_launchOnExit != nullptr && m_launchOnExit->IsEmpty() == false)
+	{		
+		try
+		{			
+			//launch the target app
+			m_launchOnExitShutdown = true;
+			auto ret = RunAsyncAndCatchErrors<bool>([&]() {
+				return create_task(Launcher::LaunchUriAsync(ref new Uri(m_launchOnExit)));
+			}, false);
+		}
+		catch (Platform::InvalidArgumentException^ e)
+		{
+		}
+	}
 }
 
 /* Application lifecycle event handlers. */
 
 void App::OnActivated(CoreApplicationView^ applicationView, IActivatedEventArgs^ args)
 {
+	//start only if not already initialized. If there is a game in progress, just return
+	if (m_initialized == true)
+	{
+		return;
+	}
+
+	int argc = NULL;
+	std::vector<char*> argv;
+	std::vector<std::string> argvTmp; //using std::string as temp buf instead of char* array to avoid manual char allocations
+	ParseProtocolArgs(args, &argc, &argv, &argvTmp);
+
+	int ret = rarch_main(argc, argv.data(), NULL);
+	if (ret != 0)
+	{
+		RARCH_ERR("Init failed\n");
+		CoreApplication::Exit();
+		return;
+	}
+	m_initialized = true;
+
+	//Should it reset drivers only if the menu is loaded, but not if the content is started directly through arguments?
```

Comment:
not resetting the driver may lead to issues where if content is only started through arguments the driver may be set to opengl and then further content loaded through arguments would also be loaded through opengl

Reply:
Got it. The problem now is that if a specific core settings are overridden to "gl", when loading content through arguments, it would start the content and the core in "gl". But this part of code would then override gl to default driver. I don't know if that's even an issue, unless there are some cores that run exclusively in "gl". Would In that case, loading through arguments most likely fail?
Is there any reason why a driver reset is forced during initialization?
I can do whatever is preferred, just trying to cover all the edge cases.

Reply:
Thanks for the info! I did some testing with Mupen64Plus-Next which is, if I understand correctly using ANGLE exclusively. If the game is started through arguments, RA will crash, because dx11 is set after the game initializes with gl. This was not a problem before, because on UWP, menu would load first and set automatically to dx11 and cores would load afterwards, so if it was gl-only it would be fine. 
Not sure what would be the best solution to make both use cases work. I would say that if the content is started from arguments, there shouldn't be any driver resetting or it should reset the driver once the content is closed and goes back to menu.

Reply:
After extensive testing, I opted for not resetting default driver only if the content is already initialized through arguments.
-------Use case 1 before fix: regular start, ANGLE content-------
1. Start RA, driver set to "dx11"
2. Change driver to "gl"
3. Load ANGLE-only content (ie. Mupen64Plus-Next)
4. Content works
5. Shut down RA
6. Start RA, driver set to "dx11"

-------Use case 2 before fix: start through arguments, ANGLE content-------
1. Start RA, loads ANGLE content, sets driver to 2gl"
2. Driver automatically reset to "dx11"
3. RA crashes
4.  Start RA, driver set to "dx11"

-------Use case 3 after fix: regular start, ANGLE content-------
Same as Use case 1

-------Use case 4 after fix: start through arguments, ANGLE content-------
1. Start RA, loads ANGLE content, sets driver to 2gl"
2. Driver not reset because content already init, checks p_content->is_inited
3. Content works
5. Shut down RA
6. Start RA, driver set to "dx11"